### PR TITLE
Add Integration for DeepInfra LLM Models

### PR DESCRIPTION
**Pull request by ovuruska**
# Description

This PR includes implementation of DeepInfra's LLMs integration to llama_index.

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [x] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [x] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [ ] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/llms/llama-index-llms-deepinfra/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+check-filenames = true
+check-hidden = true
+skip = "*.csv,*.html,*.json,*.jsonl,*.pdf,*.txt,*.ipynb"
+
+[tool.llamahub]
+contains_example = false
+import_path = "llama_index.llms.deepinfra"
+
+[tool.llamahub.class_authors]
+DeepInfraLLM = "llama-index"
```
Comment (nerdai) [inline llama-index-integrations/llms/llama-index-llms-deepinfra/pyproject.toml:15]: feel free to change `llama-index`, which is more of a placeholder, to your own Github username

**Review by nerdai — APPROVED** [review]
Thanks @ovuruska! I left some minor comments in my review.

**Comment by ovuruska** [comment]
@nerdai Thanks for the feedback

**Comment by ovuruska** [comment]
I'll handle required updates ASAP.

File: llama-index-integrations/llms/llama-index-llms-deepinfra/llama_index/llms/deepinfra/base.py [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+        Args:
+            prompt (str): The input prompt to generate completion for.
+            **kwargs: Additional keyword arguments for the API request.
+
+        Returns:
+            str: The generated text completion.
+        """
+        payload = self._build_payload(prompt=prompt, **kwargs)
+        result = self._client.request(INFERENCE_ENDPOINT, payload)
+        return CompletionResponse(text=maybe_extract_text_from_json(result), raw=result)
```
[the remark this answers is not in the document]
Comment (ovuruska) [reply llama-index-integrations/llms/llama-index-llms-deepinfra/llama_index/llms/deepinfra/base.py]: Hi @nerdai, thanks for your feedback. Currently, logprob field is not available.

File: llama-index-integrations/llms/llama-index-llms-deepinfra/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+check-filenames = true
+check-hidden = true
+skip = "*.csv,*.html,*.json,*.jsonl,*.pdf,*.txt,*.ipynb"
+
+[tool.llamahub]
+contains_example = false
+import_path = "llama_index.llms.deepinfra"
+
+[tool.llamahub.class_authors]
+DeepInfraLLM = "Oğuz Vuruşkaner <https://github.com/ovuruska>"
```
Comment (nerdai) [inline llama-index-integrations/llms/llama-index-llms-deepinfra/pyproject.toml:15]: oh please just us `ovuruska` here

**Pull request merged**